src/treelearner/serial_tree_learner.cpp

```diff
@@ -69,8 +70,6 @@ void SerialTreeLearner::Init(const Dataset* train_data, bool is_constant_hessian
   histogram_pool_.DynamicChangeSize(train_data_, config_, max_cache_size, config_->num_leaves);
   // push split information for all leaves
   best_split_per_leaf_.resize(config_->num_leaves);
-  splits_per_leaf_.resize(config_->num_leaves*train_data_->num_features());
```

Comment:
@StrikerRUS Here is an additional cost, before this PR. So I think this should be in v2.3.0 release as well

Reply:
@guolinke Sure! We have been waiting so long, so can wait a little bit more, I guess.